core/src/identity.rs

```diff
@@ -1,15 +1,20 @@
 use serde::{Deserialize, Serialize};
 
-#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
-pub struct KeyMaterial {
-    pub id: String,
-    #[serde(rename = "type")]
-    pub proof_type: String,
-    pub controller: String,
-    pub public_key_multibase: String,
+pub struct Identity {}
+
+pub async fn recover<S, T>(
+    resolver: T,
```

Comment:
Is there a reason you made this a generic vs `impl crate::DidResolver` like with the verifier? Genuinely wanted to know if there was a reason for one vs the other in this case

Reply:
So apparently they are roughly the same from an implementation perspective - https://stackoverflow.com/questions/47514930/what-are-the-differences-between-an-impl-trait-argument-and-generic-function-par

Reply:
Resolving, though I'd still love to know if you had a preference for one or the other